### PR TITLE
Feature/pow

### PR DESCRIPTION
**Pull request by Nashatyrev**
Add `DepositContract` implementation based on EthereumJ java API

### CODE REVIEW

File: pow/ethereumj/src/test/java/org/ethereum/beacon/pow/StandaloneDepositContractTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  @Test
+  public void test1() {
+    StandaloneBlockchain sb = new StandaloneBlockchain()
+        .withAutoblock(true);
+    ContractMetadata contractMetadata = new ContractMetadata();
+    contractMetadata.abi = depositAbi;
+    contractMetadata.bin = depositBin;
+    SolidityContract contract = sb.submitNewContract(contractMetadata);
+    ((SolidityContractImpl) contract).addRelatedContract(depositAbi); // TODO ethJ bug workaround
```
Comment (zilm13) [inline pow/ethereumj/src/test/java/org/ethereum/beacon/pow/StandaloneDepositContractTest.java]: new bug? could you, please, create an issue.
it was working, I thought

File: pow/ethereumj/src/test/java/org/ethereum/beacon/pow/StandaloneDepositContractTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    SolidityCallResult result = contract.callFunction(
+        Ether.valueOf(32).toGWei().times(1_000_000_000).getValue(),
+        "deposit",
+        (Object) new byte[2048]);
+
+    System.out.println(result);
+
+    ret = contract.callConstFunction("get_deposit_root");
+    System.out.println(Hex.encodeHexString((byte[]) ret[0]));
```
Comment (zilm13) [inline pow/ethereumj/src/test/java/org/ethereum/beacon/pow/StandaloneDepositContractTest.java]: maybe assert something if it's a test?

File: pow/core/src/main/java/org/ethereum/beacon/pow/AbstractDepositContract.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  private final Publisher<DepositInfo> depositStream = depositSink
+      .publishOn(Schedulers.single())
+      .doOnSubscribe(s -> depositSubscriptionsChanged())
+      .doOnCancel(() -> depositSubscriptionsChanged())
+      .onBackpressureError()
+      .name("PowClient.deposits");
+
+  private boolean depositsSubscribed;
+  private List<Deposit> initialDeposits = new ArrayList<>();
+
```
Comment (zilm13) [inline pow/core/src/main/java/org/ethereum/beacon/pow/AbstractDepositContract.java]: are you using google code style? it shouldn't like double new lines

File: pow/ethereumj/src/main/java/org/ethereum/beacon/pow/EthereumJDepositContract.java [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+      Collections.reverse(contractEvents);
+      for (Invocation contractEvent : contractEvents) {
+        if (CHAIN_START_EVENT_NAME.equals(contractEvent.function.name)) {
+          return Pair.with(block.getHash(), (byte[]) contractEvent.args[0]);
+        } else {
+          return Pair.with(block.getHash(), (byte[]) contractEvent.args[0]);
+        }
+      }
+    }
+    return null;
```
Comment (mkalinin) [inline pow/ethereumj/src/main/java/org/ethereum/beacon/pow/EthereumJDepositContract.java]: Why not an `Optional` here?

Comment (Nashatyrev) [reply pow/ethereumj/src/main/java/org/ethereum/beacon/pow/EthereumJDepositContract.java]: Kinda 'old style' with byte arrays and nulls :)
Ok, will change this

**Pull request merged**